### ISSUE TITLE
SSHUserAuthServer uses index instead of slice and fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8771]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)| @einarfd reported|
|-|-|
|Trac ID|trac#8771|
|Type|defect|
|Created|2016-08-19 12:17:14Z|
|Branch|https://github.com/twisted/twisted/tree/8771-ssh-userauth-keyboard-interactive|

In SSHUserAuthServer.ssh_USERAUTH_PK_OK_keyboard_interactive 
the code does ord(data[0]) which with a bytestring on python 3. Gives a integer and not a substring, and the ord invocation then fails with an Exception. 

<details><summary>Searchable metadata</summary>

```
trac-id__8771 8771
type__defect defect
reporter__einarfd einarfd
priority__normal normal
milestone__None None
branch__8771_ssh_userauth_keyboard_interactive 8771-ssh-userauth-keyboard-interactive
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1471609034578685 1471609034578685
changetime__1474198569965514 1474198569965514
version__None None
owner__Craig_Rodrigues__rodrigc_____ Craig Rodrigues <rodrigc@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)|@einarfd set owner to @einarfd|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)<a name="note_2"></a>|@einarfd removed owner|
|-|-|

Added a merge request for this at https://github.com/twisted/twisted/pull/507

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @rodrigc<br>@rodrigc set status to `closed`|
|-|-|

In changeset 6022f65903eda2f8df74b2d284b8ef359df24904
```
#!CommitTicketReference repository="" revision="6022f65903eda2f8df74b2d284b8ef359df24904"
Merge pull request #510 from twisted/8771-ssh-userauth-keyboard-interactive

Author: einarfd
Reviewer: rodrigc
Fixes: #8771

Fix interactive ssh prompt on Python 3
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing the review tag from closed tickets
